presto-main/src/main/java/com/facebook/presto/util/FastutilSetHelper.java

```diff
@@ -65,43 +66,140 @@ public static boolean in(boolean booleanValue, BooleanOpenHashSet set)
         return set.contains(booleanValue);
     }
 
-    public static boolean in(byte byteValue, ByteOpenHashSet set)
+    public static boolean in(double doubleValue, DoubleOpenCustomHashSet set)
     {
-        return set.contains(byteValue);
+        return set.contains(doubleValue);
     }
 
-    public static boolean in(char charValue, CharOpenHashSet set)
+    public static boolean in(long longValue, LongOpenCustomHashSet set)
     {
-        return set.contains(charValue);
+        return set.contains(longValue);
     }
 
-    public static boolean in(double doubleValue, DoubleOpenHashSet set)
+    public static boolean in(Object sliceValue, ObjectOpenHashSet<Slice> set)
     {
-        return set.contains(doubleValue);
+        return set.contains(sliceValue);
     }
 
-    public static boolean in(float floatValue, FloatOpenHashSet set)
+    public static boolean in(Object objectValue, ObjectOpenCustomHashSet<?> set)
     {
-        return set.contains(floatValue);
+        return set.contains(objectValue);
     }
 
-    public static boolean in(int intValue, IntOpenHashSet set)
+    private static final class LongStrategy
+            implements LongHash.Strategy
     {
-        return set.contains(intValue);
-    }
+        private final MethodHandle hashCodeHandle;
+        private final MethodHandle equalsHandle;
 
-    public static boolean in(long longValue, LongOpenHashSet set)
-    {
-        return set.contains(longValue);
+        private LongStrategy(FunctionRegistry registry, Type type)
+        {
+            hashCodeHandle = registry.getScalarFunctionImplementation(registry.resolveOperator(HASH_CODE, ImmutableList.of(type))).getMethodHandle();
+            equalsHandle = registry.getScalarFunctionImplementation(registry.resolveOperator(EQUAL, ImmutableList.of(type, type))).getMethodHandle();
+        }
+
+        @Override
+        public int hashCode(long value)
+        {
+            try {
+                return Ints.checkedCast((long) hashCodeHandle.invokeExact(value));
```

Comment:
actually using a raw method handle defeats the purpose of specially handling primitive types since all of the arguments and return values will be boxed.

I think this is fine for an interpreted approach, but for execution, we should definitely compile in the actual primitive calls.